src/main/java/hudson/plugins/emailext/plugins/ContentBuilder.java

```diff
@@ -68,7 +68,9 @@ public static String transformText(String origText, ExtendedEmailPublisherContex
             List<TokenMacro> macros = new ArrayList<TokenMacro>(getPrivateMacros());
             if(additionalMacros != null)
                 macros.addAll(additionalMacros);
-            newText = TokenMacro.expandAll(context.getBuild(), context.getListener(), newText, false, macros);
+            if(context.getBuild() != null) {
+                newText = TokenMacro.expandAll(context.getBuild(), context.getListener(), newText, false, macros);

```

Comment:
That's a much more difficult problem...

On Mon, Oct 26, 2015 at 5:22 PM Kanstantsin Shautsou <
notifications@github.com> wrote:

> In src/main/java/hudson/plugins/emailext/plugins/ContentBuilder.java
> <https://github.com/jenkinsci/email-ext-plugin/pull/107#discussion_r43071858>
> :
>
> > @@ -68,7 +68,9 @@ public static String transformText(String origText, ExtendedEmailPublisherContex
> >              List<TokenMacro> macros = new ArrayList<TokenMacro>(getPrivateMacros());
> >              if(additionalMacros != null)
> >                  macros.addAll(additionalMacros);
> > -            newText = TokenMacro.expandAll(context.getBuild(), context.getListener(), newText, false, macros);
> > +            if(context.getBuild() != null) {
> > +                newText = TokenMacro.expandAll(context.getBuild(), context.getListener(), newText, false, macros);
>
> As i understand that wouldn't work and i also depend on TokenMacro.expandAll(build,
> ... method. First of all TokenMacro + workflow should be resolved imho.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/email-ext-plugin/pull/107/files#r43071858>.
>